src/NUnitFramework/framework/Internal/Filters/NotFilter.cs

```diff
@@ -76,10 +76,9 @@ public override bool Match( ITest test )
         /// </summary>
         /// <param name="test">The test to which the filter is applied</param>
         /// <returns>True if the test matches the filter explicityly, otherwise false</returns>
-        // TODO: Not yet clear how explicit match should work for NotFilter
         public override bool IsExplicitMatch(ITest test)
         {
-            return !BaseFilter.IsExplicitMatch(test);
+            return !BaseFilter.Match(test);
```

Comment:
There already is one:

 

        [Test]

        public void ExplicitMatchTest()

        {

            var filter = new NotFilter(new CategoryFilter("Dummy"));

 

            Assert.True(filter.IsExplicitMatch(_topLevelSuite));

            Assert.False(filter.IsExplicitMatch(_dummyFixture));

            Assert.True(filter.IsExplicitMatch(_dummyFixture.Tests[0]));

 

            Assert.True(filter.IsExplicitMatch(_anotherFixture));

            Assert.True(filter.IsExplicitMatch(_anotherFixture.Tests[0]));

        }

 

 

From: CharliePoole [mailto:notifications@github.com] 
Sent: Saturday, October 03, 2015 2:33 PM
To: nunit/nunit
Cc: Neil Colvin
Subject: Re: [nunit] Fix NotFilter so -exclude works correctly (#862)

 

In src/NUnitFramework/framework/Internal/Filters/NotFilter.cs <https://github.com/nunit/nunit/pull/862#discussion_r41090174> :

>          public override bool IsExplicitMatch(ITest test)
>          {
> -            return !BaseFilter.IsExplicitMatch(test);
> +            return !BaseFilter.Match(test);

Need a test or tests for this. In fairness, we needed them before this, but if we're going to fix it then let's have some tests that show when a NotFilter match is considered Explicit and when it isn't.

—
Reply to this email directly or view it on GitHub <https://github.com/nunit/nunit/pull/862/files#r41090174> .  <https://github.com/notifications/beacon/AFh64ZsMT-42CDlj1S7TzvdKihSNBpnjks5o4BbKgaJpZM4GIcgJ.gif>